aspnetcore/signalr/configuration.md

```diff
@@ -0,0 +1,202 @@
+            options.PayloadSerializerSettings.ContractResolver = 
+            new DefaultContractResolver();
+        });
+}
+```
+
+On a .NET client, the same `AddJsonHubProtocol` extension method exists on `HubConnectionBuilder`. The `Microsoft.Extensions.DependencyInjection` namespace must be imported to see the method:
+
+```csharp
+// At the top of the file:
+using Microsoft.Extensions.DependencyInjection;
+
+// When constructing your connection:
+var connection = new HubConnectionBuilder();
+    .AddJsonHubProtocol(options => {
+        options.PayloadSerializerSettings.ContractResolver = 
+            new DefaultContractResolver();
+    });
+```
+
+### MessagePack Serialization Options
+
+MessagePack serialization can be configured by providing a delegate to the `AddMessagePackProtocol` call. See [MessagePack in SignalR](xref:signalr/messagepack) for more details.
+
+> [!NOTE]
+> It's not possible to configure JSON or MessagePack serialization in the JavaScript client at this time.
+
+### Configure [HubOptions](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.signalr.huboptions) for SignalR
+
+The following table describes the `HubOptions` options for configuring the hub:
+
+| Option | Description |
+| ------ | ----------- |
+| `HandshakeTimeout` | If the client doesn't send an initial handshake message within this time interval, the connection will be closed |
+| `KeepAliveInterval` | If the server hasn't sent a message within this interval, a ping message will is sent automatically to keep the connection open |
+| `SupportedProtocols` | Protocols supported by this hub. By default, all protocols registered on the server are allowed, but protocols can be removed from this list to disable specific protocols for individual hubs. |
+| `EnableDetailedErrors` | If true, sends detailed error messages to the client when exceptions occur. This may contain sensitive data and is `false` by default |
+
+The `HubOptions` are configured in the `ConfigureServices` method of the `Startup` class, as shown in the following code sample:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddSignalR(hubOptions =>
+    {
+        hubOptions.EnableDetailedErrors = true;
+        hubOptions.KeepAliveInterval = TimeSpan.FromMinutes(1);
+    })
+}
+```
+
+Strongly-typed [HubOptions<T>](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.signalr.huboptions-1) can be configured as follows, for example, setting the `EnableDetailedErrors` property:
+
+```csharp
+  services.AddSignalR().AddHubOptions<HubName>(options =>
+  {
+      options.EnableDetailedErrors = true;
+  }
+```
+
+### HttpConnectionDispatcherOptions
+
+Options related to the transport layer. Use these to restrict the transports that can be used by SignalR clients, as well as to configure advanced settings related to memory buffer management. These options are configured by passing a delegate to `MapHub<T>`.
```

Comment:
>Options related to the transport layer.

That doesn't make sense by itself. SHould it be

### HttpConnectionDispatcherOptions - Options related to the transport layer.